apps/dashboard/app/helpers/batch_connect/session_contexts_helper.rb

```diff
@@ -136,4 +136,9 @@ def resolution_field(form, id, opts = {})
       )
     end
   end
+
+  def parse_favorites(favorites)
+    return favorites if favorites.any? { |f| f.is_a?(FavoritePath) }
+    favorites.map { |f| FavoritePath.new(f) }
+  end
```

Comment:
Can we put all the logic in this function? Something like this below (the name change is a bit of a nitpick I'll admit, but there's something I don't like about `parse`. I say this because I think we have to account for a few things, and we may as well co-locate all that logic.

* if they configure `favorites: false` it won't show favorites at all (we can do this later in another ticket)
* I'm pretty sure we shouldn't be using remote file systems. I just don't know how they could be _actually_ useful here
given the script would likely have to mount them to a real filesystem location or something? So we'd have to fallback to `OodFilesApp.new.favorite_paths.reject(:&remote?)`

```ruby
def pathselector_favorites(config)

end
```

so that 

```ruby
favorites = field_options.fetch(:favorites, OodFilesApp.new.favorite_paths)
```

just becomes
```
favorites = pathselector_favorites(field_options.fetch(:favorites, []))
```